src/helpers/figures/charts/bar_chart.ts

```diff
@@ -234,7 +235,21 @@ function getBarConfiguration(
     horizontalChart: chart.horizontal,
   });
   const legend: DeepPartial<LegendOptions<"bar">> = {
-    labels: { color: fontColor },
+    ...INTERACTIVE_LEGEND_CONFIG,
+    labels: {
+      color: fontColor,
+      usePointStyle: true,
+      //@ts-ignore
+      generateLabels: (_chart) =>
+        _chart.data.datasets.map((dataset, index) => ({
+          text: dataset.label ?? "",
+          strokeStyle: dataset.borderColor,
```

Comment:
Tis thing also seems to be repeated 5 times. A helper would also probaly allo to hide the ts-ignore inside  it